src/go/rpk/pkg/api/admin/api_cloud_storage.go

```diff
@@ -41,6 +42,25 @@ type TopicRecoveryStatus struct {
 	RecoveryRequest RecoveryRequestParams `json:"request"`
 }
 
+// CloudStorageStatus represents the status of a partition of a topic in the
+// cloud storage.
+type CloudStorageStatus struct {
+	CloudStorageMode      string `json:"cloud_storage_mode"`                    // The partition's cloud storage mode (one of: disabled, write_only, read_only, full and read_replica).
+	LastManifestUpload    int    `json:"ms_since_last_manifest_upload"`         // Delta in milliseconds since the last upload of the partition's manifest.
+	LastSegmentUpload     int    `json:"ms_since_last_segment_upload"`          // Delta in milliseconds since the last segment upload for the partition.
+	LastManifestSync      *int   `json:"ms_since_last_manifest_sync,omitempty"` // Delta in milliseconds since the last manifest sync (only present for read replicas).
```

Comment:
Can these be `MsSinceLast...`? When looking at the usage below, I was confused how we were using a timestamp to find the latest, but these aren't timestamps.